### PR TITLE
metacubexd: init at 1.140.0

### PR DESCRIPTION
**Pull request by Guanran928**
## Description of changes

Clash/Mihomo WebUIs, used with https://github.com/NixOS/nixpkgs/pull/287966

*sigh*, bun and pnpm.

https://github.com/MetaCubeX/metacubexd
https://github.com/haishanh/yacd

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/me/metacubexd/package.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out
+    cp -r ./* $out
+
+    runHook postInstall
+  '';
+
+  passthru.updateScript = nix-update-script { };
```
Comment (Aleksanaa) [inline pkgs/by-name/me/metacubexd/package.nix:28]: Won't work in this situation?

**Review by Aleksanaa — APPROVED** [review]
Others looks okay

File: pkgs/by-name/me/metacubexd/package.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+, stdenvNoCC
+, fetchurl
+, nix-update-script
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "metacubexd";
+  version = "1.138.1";
+
+  src = fetchurl {
+    url = "https://github.com/MetaCubeX/metacubexd/releases/download/v${finalAttrs.version}/compressed-dist.tgz";
```
Comment (SuperSandro2000) [inline pkgs/by-name/me/metacubexd/package.nix:11]: Why not build from source?

File: pkgs/by-name/ya/yacd/package.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+, fetchurl
+, nix-update-script
+, stdenvNoCC
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "yacd";
+  version = "0.3.8";
+
+  src = fetchurl {
+    url = "https://github.com/haishanh/yacd/releases/download/v${finalAttrs.version}/yacd.tar.xz";
```
Comment (SuperSandro2000) [inline pkgs/by-name/ya/yacd/package.nix:11]: same here

**Comment by SuperSandro2000** [comment]
> _sigh_, bun and pnpm.

With #290715 merged, does that make things easier and we can build from source?

File: pkgs/by-name/me/metacubexd/package.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    runHook preInstall
+
+    cp -r ./dist $out
+
+    runHook postInstall
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
```
Comment (drupol) [inline pkgs/by-name/me/metacubexd/package.nix:48]: No `meta.mainProgram` ?

Comment (drupol) [reply pkgs/by-name/me/metacubexd/package.nix:48]: Fair enough. going to merge it.

**Pull request merged**